luigi_pipeline/lib/model/gcnv_mt_schema.py

```diff
@@ -0,0 +1,212 @@
+import datetime
+import re
+
+import hail as hl
+
+from lib.model.base_mt_schema import BaseMTSchema, row_annotation
+from lib.model.seqr_mt_schema import SeqrGenotypesSchema, SeqrVariantsAndGenotypesSchema
+
+from hail_scripts.computed_fields import variant_id
+
+SAMPLE_ID_REGEX = r'(.+)_v\d+_Exome_(C|RP-)\d+$'
+
+def parse_genes(gene_col: hl.expr.StringExpression) -> hl.expr.SetExpression:
+    """
+    Convert a string-ified gene list to a set()
+    """
+    return hl.set(hl.map(
+        lambda gene: gene.split(r'\.')[0],
+        hl.filter(
+            lambda gene: ~hl.set({'None', 'null', 'NA', ''}).contains(gene),
+            gene_col.split(','),
+        )
+    ))
+
+def hl_agg_collect_set_union(gene_col: hl.expr.SetExpression) -> hl.expr.SetExpression:
+    """
+    aggregate with the set union operator
+    """
+    return hl.fold(
+        lambda i, j: i | j,
+        hl.empty_set(hl.tstr),
+        hl.agg.collect(gene_col),
+    )
+
+class SeqrGCNVVariantSchema(BaseMTSchema):
+
+    @row_annotation()
+    def contig(self):
+        return self.mt.chr.replace("^chr", "")
+
+    @row_annotation()
+    def sc(self):
+        return self.mt.vac
+
+    @row_annotation()
+    def sf(self):
+        return self.mt.vaf
+
+    @row_annotation()
+    def sn(self):
+        return hl.if_else(
+            hl.is_defined(self.mt.vaf),
+            hl.int(self.mt.vac / self.mt.vaf),
+            hl.missing(hl.tint32)
+        )
+
+    @row_annotation(name='svType')
+    def sv_type(self):
+        return self.mt.svtype
+
+    @row_annotation(name='StrVCTVRE_score')
+    def strvctvre(self):
+       return self.mt.strvctvre_score
+
+    @row_annotation(name='variantId')
+    def variant_id(self):
+        return hl.format(f"%s_%s_{datetime.date.today():%m%d%Y}", self.mt.variant_name, self.mt.svtype)
+
+    @row_annotation()
+    def start(self):
+        return hl.agg.min(self.mt.sample_start)
+
+    @row_annotation()
+    def end(self):
+        return hl.agg.max(self.mt.sample_end)
+
+    @row_annotation()
+    def num_exon(self):
+        return hl.agg.max(self.mt.genes_any_overlap_totalExons)
+
+    @row_annotation(name='geneIds')
+    def gene_ids(self):
+        return hl.array(hl_agg_collect_set_union(parse_genes(self.mt.genes_any_overlap_Ensemble_ID)))
+
+    @row_annotation(name='transcriptConsequenceTerms', fn_require=sv_type)
+    def transcript_consequence_terms(self):
+        formatted_gcnv_consequence = hl.format('gCNV_%s', self.mt.svType)
+        lof = hl.len(hl_agg_collect_set_union(parse_genes(self.mt.genes_LOF_Ensemble_ID))) > 0
+        copy_gain = hl.len(hl_agg_collect_set_union(parse_genes(self.mt.genes_CG_Ensemble_ID))) > 0
```

Comment:
You are already computing these consequence lists below in `sorted_transcript_consequences`. It would be neater to make this function depend on that function and then compute this list based on the values of `major_consequence` in that field

Reply:
Yep that makes sense!  I've been a bit wary about adding fields that we don't have (want?) in elastic search.  Is that generally ok?  Is `disable_index` useful here?

Reply:
I'm a bit confused by what would be added to elasticsearch here - we want both the `sorted_transcript_consequences` and  `transcript_consequence_terms` to run and populate in elsaticsearch. The way you have the code now runs these functions in parallel, and I am proposing you run these functions in sequence so that one can be computed from the other to prevent duplicating logic

Reply:
Ha.  I was answering a different question than you asked sorry.  My original implementation of this had `hl_agg_collect_set_union(parse_genes(self.mt.genes_LOF_Ensemble_ID)` and `hl_agg_collect_set_union(parse_genes(self.mt.genes_CG_Ensemble_ID))` as subfields just used those on a second round of annotations.  I backed away because I wasn't sure how additional fields are handled upstream.

I'm following what you're saying though and have implemented `transcript_consequence_terms` to depend on `sorted_transcript_consequences`